vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomActivity.kt

```diff
@@ -69,20 +69,35 @@ class CreateRoomActivity : VectorBaseActivity<ActivitySimpleBinding>(), ToolbarC
                 .subscribe { sharedAction ->
                     when (sharedAction) {
                         is RoomDirectorySharedAction.Back,
-                        is RoomDirectorySharedAction.Close -> finish()
+                        is RoomDirectorySharedAction.Close             -> finish()
+                        is RoomDirectorySharedAction.CreateRoomSuccess -> {
+                            setResult(Activity.RESULT_OK, Intent().apply { putExtra(RESULT_CREATED_ROOM_ID, sharedAction.createdRoomId) })
+                            finish()
+                        }
+                        else                                           -> {
+                            // nop
+                        }
                     }
                 }
                 .disposeOnDestroy()
     }
 
     companion object {
-        private const val INITIAL_NAME = "INITIAL_NAME"
-        private const val IS_SPACE = "IS_SPACE"
 
-        fun getIntent(context: Context, initialName: String = "", isSpace: Boolean = false): Intent {
+        const val RESULT_CREATED_ROOM_ID = "RESULT_CREATED_ROOM_ID"
```

Comment:
I would prefer to keep that const val private and add a fun in the companion to get the result, for instance like here https://github.com/vector-im/element-android/blob/develop/vector/src/main/java/im/vector/app/features/spaces/SpaceCreationActivity.kt#L178

Reply:
Done